sim/gui.py

```diff
@@ -293,6 +281,27 @@ def draw_maze(
             if Walls.WEST in walls:
                 pg.draw.line(self.screen, wall_color, (x, y - line_end), (x, y + line_end + tile_size), self.wall_thickness)
 
+        # Draw robot's route after all other maze elements:
+        for i, cell in enumerate(maze.route):
```

Comment:
For symmetry, I don't like the mix of both an index and an item.

Alternatives:

```suggestion
        for begin, end in zip(maze.route, maze.route[1:]):
```

Or

```suggestion
        for i in range(len(maze.route) - 1):  # `- 1` eliminates the `if`
```